agent/php_wrapper.c

```diff
@@ -73,6 +73,26 @@ nruserfn_t* nr_php_wrap_user_function_before_after_clean(
   return wraprec;
 }
 
+nruserfn_t* nr_php_wrap_user_function_before_after_clean_extra(
+    const char* name,
+    size_t namelen,
+    nrspecialfn_t before_callback,
+    nrspecialfn_t after_callback,
+    nrspecialfn_t clean_callback,
+    const char *extra) {
+
+  nruserfn_t* wraprec = nr_php_add_custom_tracer_named(name, namelen);
+
+  wraprec->extra = extra;
+
+  nr_php_wraprec_add_before_after_clean_callbacks(name, namelen, wraprec,
+                                                  before_callback,
+                                                  after_callback,
+                                                  clean_callback);
+
+  return wraprec;
```

Comment:
This avoids duplicating logic of `nr_php_wrap_user_function_before_after_clean` and ensures `wraprec->extra` is set only if `wraprec` is not `NULL`:
```suggestion
  nruserfn_t* wraprec = nr_php_wrap_user_function_before_after_clean(
      name, namelen, before_callback, after_callback, clean_callback);
  if (nrunlikely(NULL == wraprec)) {
    nrl_warning(NRL_INSTRUMENT, "%s: unable to wrap '%s'", __func__,
                NRSAFESTR(name));
    return wraprec;
  }

  wraprec->extra = extra;
  return wraprec;
```

Reply:
f3164c6d31bcd5a4ea17d20d6c4d43983ed3018d